WEB-308: fix hardcoded documentation links

shubhamkumar9199 opened the pull request:
This PR centralizes all documentation URLs behind a DocumentationLinksService and a documentationLink pipe, backed by a shared map of documentation paths. This removes hard-coded wiki URLs from templates and help actions and makes documentation links configurable per environment.

[WEB-308](https://mifosforge.jira.com/browse/WEB-308)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] If you have multiple commits please combine them into one commit by squashing them.

- [X] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.


[WEB-308]: https://mifosforge.jira.com/browse/WEB-308?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Adds a configurable documentation base URL (environment setting) for documentation links.
  * Adds a template-friendly documentation link pipe to resolve doc pages.

* **Refactor**
  * Replaces hard-coded external help/documentation URLs throughout the UI with dynamic documentation links.
  * Centralizes opening of documentation via a shared link service used by header/help actions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shubhamkumar9199 committed:
WEB-308: fix hardcoded documentation links

fix: Add security attributes to all external documentation links

Fix:added the missing mapping in env.template.js

Removed DocumentationLinkPipe from PipesModule

alberto-art3ch reviewed (changes requested):
It is not necessary to remove those links, just to be sure that we are using the tag attribute `rel="noopener noreferrer"`  and the target to blank `target="_blank"` 

ie:
`<a href="https://mifosforge.jira.com/wiki/spaces/docs/pages/67895316/Closing+Entries " rel="noopener noreferrer"  `

shubhamkumar9199 commented:
> It is not necessary to remove those links, just to be sure that we are using the tag attribute `rel="noopener noreferrer"` and the target to blank `target="_blank"`
> 
> ie: `<a href="https://mifosforge.jira.com/wiki/spaces/docs/pages/67895316/Closing+Entries " rel="noopener noreferrer" `

@alberto-art3ch I've fixed the issue. Added rel="noopener noreferrer" to all external documentation links with target="_blank"

alberto-art3ch reviewed (approved):
LGTM
